### PR TITLE
[issue_tracker] Enable editing of site, assignee and watchers in Batch Mode

### PR DESCRIPTION
**Pull request by ay-bh**
## Brief summary of changes
- Added site selector to Issue Card edit mode
- Enhanced Add Comment modal with assignee and watchers selectors
- Updated backend endpoint to fetch otherWatchers and assignees

#### Testing instructions
1. Go to Issue Tracker Batch Mode
2. Click "Edit Issue" on any issue
3. Verify site can be modified using the new dropdown
4. Update Issue
5. Click "Add Comment"
6. Verify assignee and watchers can be modified in the modal
7. Submit changes and verify they persist

#### Link(s) to related issue(s)
* Part of #9418 

#### Screenshots
image
image

### CODE REVIEW

**Comment by christinerogers** [comment]
@ay-bh if this is ready for review -- please request review from @regisoc and @racostas

**Review by regisoc — COMMENTED** [review]
Thanks for this PR, comments:
- Some static test did not pass.
- adding/removing watchers looks good and in sync when switching from edit/detail view to batch mode.
- site dropdown, I tried to change to `No Site` but had the following image show up. 
This option should not be added to force user to choose one site.

image

- issue: unassigned state card text. When no assignee or unassigning someone from an issue, there is a blank in the card. This should instead have an explicit "unassigned" tag. Also, on a second thought, it is in line with the previous comment, not allowing to have a ticket unassigned might be a good idea. The goal of issue tracker is to resolve those tickets. Not having someone assigned, they will be forgotten.

image

- issue maybe? @racostas I think it is not linked to this PR: when updating the assignee, no history about this change in the edit/detailed view. The change is acted on the batch mode card view, and in the edit/detailed view but not in the history section.

image

**Review by regisoc — COMMENTED** [review]
Thanks for this update @ay-bh :
- [x] `No site` issue: is ok, option removed.
- [ ] `Unassigned` Issue : states `Assignee:None` in card. Front-end looks ok. I am scratching my head here as I see it as mandatory as the previous point. In HBCD, we enforced a default assignee based on category so that issues do not fall into limbo. @christinerogers  @racostas does that make sense to you that an issue is not assigned? 
- [ ] `no assignee update in history` issue: could/should be referenced in another issue ticket if necessary.

@racostas @christinerogers If the second point is ok to you, then the PR looks ok to me.

**Comment by christinerogers** [comment]
Thanks @regisoc for these comments. 
from the bottom up: 
> * [ ]  `no assignee update in history` issue: could/should be referenced in another issue ticket if necessary.
Agree this should be another issue, and it's important.   I'll add this to the Followup ticket #9418.   @ay-bh if you feel keen to address this on your own time, thank you - it's optional since you have completed GSOC. 

> * [x]  `No site` issue: is ok, option removed.
> * [ ]  `Unassigned` Issue : states `Assignee:None` in card.
@regisoc I get why HBCD has customized this module to make assignee and site selection mandatory, and there are valid reasons these were not mandatory in the Loris module design.  
Let's leave this as it is currently, keep the no-site enforcement change, and I'll add a follow up point in #9418 to make both of these configurable parameters. 

> @racostas @christinerogers If the second point is ok to you, then the PR looks ok to me.

Agreed let's proceed with this PR as is. 

Note this no-site design change requested by regis should be documented in the module readme, and this is one of many points to follow up on in #9418

**Comment by regisoc** [comment]
@racostas waiting for your review, I will put you as assignee.

**Review by racostas — APPROVED** [review]
LGTM.

I reviewed all again and manual tested:
- Functionalities looks good.
- Assignee not been updated in history I checked and effectively is a preexisting issue. I agree we should tackle this. Yes, let's create an issue for it ... or we can added it to https://github.com/aces/Loris/issues/9418 since this look a very good summary ticket for all `issue_tracker` related developments. Do you think this could be a good idea @regisoc ?
- Regarding making the assignee mandatory, I don't have a super strong opinion, I see for example in Redmine is not mandatory, but something we can think about it. It could be a new point in #9418 as well.

In conclusion, I think this PR is good to go. I leave @driusan the final review / approval.

**Comment by racostas** [comment]
I saw the two point were already added to #9418 :)
All good !
